### PR TITLE
Content directory CLI functions

### PR DESCRIPTION
**Pull request by jessebmiller**
I don't have it all hooked up to the command line yet, but the functions are written and tested

### CODE REVIEW

File: cli/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+import fetch from 'cross-fetch'
+
+import { getConfig, getSecrets } from '@squad/lib'
+import { ethers, Contract } from 'ethers'
+
+import * as fs from 'fs'
+
+// TODO move TxResponse interface from tests to lib
+interface TxResponse {
```
Comment (jessebmiller) [inline cli/src/index.ts:26]: Use ether's tx response type?

File: cli/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+        uri: '/ens/ipfs.web3api.eth',
+        plugin: ipfsPlugin({
+          provider: ipfs
+        })
+      }
+    ]
+  })
+}
+
+const squadUri = '/ens/testnet/squadprotocol.eth'
```
Comment (jessebmiller) [inline cli/src/index.ts:110]: probably want this in the config

File: cli/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+// TODO support standard wallet storage format
```
Comment (jessebmiller) [inline cli/src/index.ts:112]: todo support hardware wallets

File: cli/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+  nftId?: number
+  id?: string
+}
+
+export async function content ({
+  first = 100,
+  skip = 0,
+  type,
+  nftAddress,
+  nftId,
```
Comment (jessebmiller) [inline cli/src/index.ts:320]: remove this

File: cli/test/content.spec.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    )
+    expect(ethers.BigNumber.from(id.split('-')[1] ?? '').gte(0)).toBe(true)
+  })
+
+  test('returns content by nftAddress', async () => {
+    const got = await content({
+      nftAddress: config.contracts.ERC721Squad.address
+    })
+    const lines = got.split('\n')
+    expect(lines.length).toBeGreaterThan(3)
```
Comment (jessebmiller) [inline cli/test/content.spec.ts:89]: should be more than 2

File: cli/src/index.ts (hunk 1 above)
Comment (jessebmiller) [reply cli/src/index.ts:26]: done

File: cli/src/index.ts (hunk 2 above)
Comment (jessebmiller) [reply cli/src/index.ts:110]: moved

File: cli/test/content.spec.ts (hunk 5 above)
Comment (jessebmiller) [reply cli/test/content.spec.ts:89]: reduced

**Comment by jessebmiller** [comment]
@EzraWeller comments resolved, and early UI added. Maybe we want another quick review?

File: cli/dist/src/index.js [hunk 6]
```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env node
```
Comment (EzraWeller) [inline cli/dist/src/index.js:1]: Do we want to be committing the dist here?

File: cli/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+    console.error(
+      `No private key found in ~/.squad/${config.network as string}-secrets.json.`
+    )
+    console.error('using new random key')
+    return ethers.Wallet.createRandom()
+  }
+  return new ethers.Wallet(secrets.cliPrivateKey)
+}
+
+// TODO support data files in CLI
```
Comment (EzraWeller) [inline cli/src/index.ts:126]: What do you mean "data files"? Something other than the filetypes of 135-140, I assume?

File: cli/dist/src/index.js (hunk 6 above)
Comment (jessebmiller) [reply cli/dist/src/index.js:1]: Nope will fix

**Review by EzraWeller — APPROVED** [review]
Didn't read every line, but looks good!

File: cli/src/index.ts (hunk 7 above)
Comment (jessebmiller) [reply cli/src/index.ts:126]: Right now it only supports the lsit of text files starting on line 134, we should add raw data file formats too

**Pull request merged**